homework/week4/chrispark0304/4673.c

```diff
@@ -0,0 +1,33 @@
+#include <stdio.h>
+
+int self_num(int n){
+    int self_n = n;
+    while(n > 0){
+        self_n += n % 10;
+        n /= 10;
+    }
+    return self_n;
+}
+
+void find_self_num(int num_set[], int n){ //n에서부터 시작되는 d(n)을 10000이하에서 찾기
+    int d = self_num(n);
+    if(d > 10000){
+        return;
+    }
+    num_set[self_num(n)] = 1;
+    find_self_num(num_set , self_num(n));
```

Comment:
self_num이 여러번 호출되고 있습니다. 이미 위에서 변수를 받았기 때문에, 아래처럼 변경하시는건 어떠신가요?
```suggestion
    int d = self_num(n);
    if(d > 10000){
        return;
    }
    num_set[d] = 1;
    find_self_num(num_set , d);
```